prime-router/metadata/HL7/catchall/hl7/segments/IN1/Coverage.yml

```diff
@@ -2,47 +2,94 @@
 
 resourceType: Coverage
 
+# IN1.2 CWE[Identifier] is a subset of the input CWE; maps identifier.name which does not exist; use IN1Extension
+# IN1.4 references IN1.5 for address but both fields are repeatable; use IN1Extension for these fields
+# IN1.10 and IN1.49 are mapped to the same FHIR extension; use IN1Extension for these fields
+# IN1.10 when CX.5 is not "SN" has a definition that conflicts with XON[Organization] with no clear hierarchy
+# IN1.11 is repeatable but maps to policyHolder on Coverage which is not, so it's all captured in IN1Extension
+# IN1.16 is repeatable but maps to subscriber on Coverage which is not, so it's all captured in IN1Extension
+
 id:
   type: STRING
   valueOf: "GeneralUtils.generateResourceId()"
   expressionType: JEXL
 
-extension:
+payor:
+  expressionType: reference
+  condition: $in14 NOT_NULL || $in15 NOT_NULL
+  valueOf: datatypes/XON/Organization
+  specs: IN1.4
+  vars:
+    XADAddress: IN1.5
+    in14: STRING_ALL, IN1.4
+    in15: STRING_ALL, IN1.5
+
+period:
+  valueOf: datatypes/DT/Period
+  expressionType: resource
+  vars:
+    start: IN1.12
+    end: IN1.13
+
+policyHolder:
+  expressionType: reference
+  condition: $in111 NOT_NULL
+  valueOf: datatypes/XON/Organization
+  specs: IN1.11
+  vars:
+    in111: STRING_ALL, IN1.11
+
+relationship:
+  expressionType: resource
+  valueOf: datatypes/CWE/CodeableConcept
```

Comment:
I don't think I've seen such functionality on the HL7 > FHIR library. I think we'd have to do it as a sender transform, but I also think this is a bigger ticket where we have to also decide how to handle cases where this mapping would cause data loss.

Reply:
Apparently I am just not aware of how we handle ValueSet mappings. From what I remember the answer is that we usually try not to, but there are definitely a few places where we are doing it but I cannot figure out the mechanism for it. 

I'll bring this up at the next office hours and don't think it should hold up this pr. I'll take it as a todo to bring up and log a follow up if need be. 

Just to document it somewhere, here is one example from our code base:
[Encounter.status has a required binding. ](http://hl7.org/fhir/R4/encounter.html)
Mapping inventory does have value set mappings for it. [[1](https://docs.google.com/spreadsheets/d/1Sqt8f_LpBVMIdJxcAlB4emDOBqXV8rhy60V2_yaDp7c/edit?gid=0#gid=0), [2](https://docs.google.com/spreadsheets/d/1-g6mrN1Q_Y_foJjlOXY32UAtMWwPi245mh3kRc8FBQ4/edit?gid=0#gid=0)]
And we do get these mapped values from somewhere as it's [in some of the fhir](https://github.com/CDCgov/prime-reportstream/blob/345f93ccd4561677350297d27eae4b9ac006f3ef/prime-router/src/testIntegration/resources/datatests/mappinginventory/catchall/pv1pv2/PV1-to-Encounter.fhir#L401) we've generated for tests.